src/physics/transform.ts

```diff
@@ -74,15 +82,18 @@ export function identityFrame(out: Frame) {
 }
 
 // TRANSFORM
-export const TransformDef = EM.defineComponent("transform", (t?: mat4) => {
-  return t ?? mat4.create();
-});
+export const TransformDef = EM.defineComponent(
+  "transform",
+  () => mat4.create(),
+  (p, t?: mat4.InputT) => (t ? mat4.copy(p, t) : p)
+);
 export type Transform = mat4;
 
 // POSITION
 export const PositionDef = EM.defineComponent(
   "position",
-  (p?: vec3) => p || V(0, 0, 0)
+  () => V(0, 0, 0),
+  (p, v?: vec3.InputT) => (v ? vec3.copy(p, v) : p)
```

Comment:
Here's perhaps the most important change in the whole thing:
we can now use EM.set(myEntinty, PositionDef, [1,0,1]) and it ensures the component exists and copies in [1,0,1] either way